src/contexts/TotalStaked/index.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { Signer } from 'ethers';
+
+import { TOKEN_GEYSER_ADDRESS } from 'constants/index';
+
+import { Web3Context } from 'contexts/Web3';
+
+import { getTokenGeyserContract } from 'utils';
+
+export interface ContextValues {
+  // eslint-disable-next-line
+  totalStaked?: any;
+}
+
+export const TotalStakedContext = React.createContext<ContextValues>({});
+
+export const TotalStakedProvider: React.FC = ({ children }) => {
+  const { address, signer } = React.useContext(Web3Context);
+  const [totalStaked, setTotalStaked] = React.useState<string>();
+
+  const fetchTotalStaked = React.useCallback(async (userAddress: string, signer: Signer) => {
+    const geyserContract = getTokenGeyserContract(signer, TOKEN_GEYSER_ADDRESS as string);
+    try {
+      const balance = await geyserContract.totalStakedFor(userAddress);
+      setTotalStaked(balance.toString());
+    } catch (e) {
+      console.error(e);
+      setTotalStaked('0');
+    }
+  }, []);
+
+  React.useEffect((): (() => void) => {
+    if (address && signer) {
+      fetchTotalStaked(address, signer);
+      const refreshInterval = setInterval(() => fetchTotalStaked(address, signer), 10000);
+      return () => clearInterval(refreshInterval);
+    } else {
+      setTotalStaked('0');
```

Comment:
I was also wondering, is there a possibility here of a race condition affecting the stake info, something like going to another view and coming back before the ten seconds of retry are up? I haven't thought about it too much but maybe it's the case. If we wanna be really really safe, implementing a basic `cancelled/retry` mechanism would be cool

Reply:
Just tested and doesn't appear to be an issue! If I were to implement a basic `cancelled/retry` mechanism, how would one do it in this context? My first thought is just running the `refreshInterval` in the `catch` part of my `try/catch` (?)

Reply:
Yup! But I think here it will not be necessary :).